zebrad/src/components/sync/recent_sync_lengths/tests/prop.rs

```diff
@@ -0,0 +1,65 @@
+//! Randomised property tests for recent sync lengths.
+
+use proptest::prelude::*;
+use proptest_derive::Arbitrary;
+
+use super::super::RecentSyncLengths;
+
+#[derive(Arbitrary, Clone, Copy, Debug)]
+enum SyncLengthUpdate {
+    ObtainTips(usize),
+    ExtendTips(usize),
+}
+
+use SyncLengthUpdate::*;
+
+proptest! {
+    #[test]
+    fn max_recent_lengths(
+        sync_updates in any::<Vec<SyncLengthUpdate>>(),
+    ) {
+        let (mut recent_sync_lengths, receiver) = RecentSyncLengths::new();
+
+        for update in sync_updates {
+            match update {
+                ObtainTips(sync_length) => recent_sync_lengths.push_obtain_tips_length(sync_length),
+                ExtendTips(sync_length) => recent_sync_lengths.push_extend_tips_length(sync_length),
+            }
+        }
+
+        prop_assert!(
+            receiver.borrow().len() <= RecentSyncLengths::MAX_RECENT_LENGTHS,
+            "recent sync lengths: {:?}",
+            *receiver.borrow(),
+        );
```

Comment:
Optional idea: a stricter assertion :thinking: 

```suggestion
        let (mut recent_sync_lengths, receiver) = RecentSyncLengths::new();

        for (count, update) in sync_updates.into_iter().enumerate() {
            match update {
                ObtainTips(sync_length) => recent_sync_lengths.push_obtain_tips_length(sync_length),
                ExtendTips(sync_length) => recent_sync_lengths.push_extend_tips_length(sync_length),
            }

            prop_assert_eq!(
                receiver.borrow().len(),
                count.min(RecentSyncLengths::MAX_RECENT_LENGTHS),
                "recent sync lengths: {:?}",
                *receiver.borrow(),
            );
        }
```